blockchain/blockchain.go

```diff
@@ -369,6 +380,30 @@ func verifyBlock(txn db.Transaction, block *core.Block) error {
 		return ErrParentDoesNotMatchHead
 	}
 
+	head, err := l1Head(txn)
+	switch {
+	case err == nil:
+		var header *core.Header
+		if head.BlockNumber == block.Number {
+			header = block.Header
+		} else {
+			header, err = b.BlockHeaderByNumber(head.BlockNumber)
+		}
+
+		if err == nil && !head.BlockHash.Equal(header.Hash) {
```

Comment:
I am not sure if I follow this.

We are checking if `head.BlockHash.Equal(header.Hash)` but `head.BlockNumber == header.Number` is not necessarily always true. So how do we expect `head.BlockHash.Equal(header.Hash)` to be always equal?

Reply:
> `head.BlockNumber == header.Number` is not necessarily always true

We get the matching header from the db in the `else` clause a few lines above.